packages/pyright-internal/src/languageService/completionProvider.ts

```diff
@@ -2497,13 +2497,22 @@ export class CompletionProvider {
         });
     }
 
-    private _addNamedParametersToMap(type: FunctionType, paramMap: Map<string, string>) {
+    private _addNamedParametersToMap(type: FunctionType, names: Set<string>) {
         type.details.parameters.forEach((param) => {
-            if (param.name && !param.isNameSynthesized) {
+            if (
+                param.category === ParameterCategory.VarArgDictionary &&
```

Comment:
this condition seems duplicated here and in printer. can we share?

Reply:
Yeah I think so. I'll do that.